symbol.c

```diff
@@ -983,7 +958,7 @@ rb_sym2id(VALUE sym)
 
             if (UNLIKELY(!(id & ~ID_SCOPE_MASK))) {
                 VALUE fstr = RSYMBOL(sym)->fstr;
-                ID num = next_id_base_with_lock(symbols);
+                ID num = next_id_base();
```

Comment:
Also `next_id_base` shouldn’t have the `symbols` argument?

Reply:
We don't need it anymore because we can just use the `ruby_global_symbols` global variable since `next_id_base` does not require the VM lock anymore.